### ISSUE TITLE
Feature Request: Prometheus Endpoint for Large Scale Monitoring 

### ISSUE THREAD

**Issue by liskl** [ethereum-mining/ethminer#124]
I'd love to see a Prometheus Endpoint for Eth Miner hosted on the microhttpd interface maybe

/metrics route

that can give us more time based stats for the service using just a reqular http plain/text based output page.

examples of howto display the text so prometheus can pull it easily
https://prometheus.io/docs/instrumenting/exposition_formats/

would be willing to pitch a few Etherium to whoever just gets the following instrumented on the http://0.0.0.0:3333/metrics route

```# HELP hashes_Per_Second The Current Hashes per second 
# TYPE hashes_Per_Second Gauge
hashes_Per_Second 24770000
shares_Accepted_Total_Since_Start 200```

**Comment by smurfy** [comment]
@chfast 
Wanted to start working on a remote monitoring but i ran into various things.
* libjsonrpccpp has no http_server enabled (missing microhttpd)
* libjsonrpccpp tcp socket server fails to compile on windows
* microhttpd  not yet a hunter module (i saw you are working on that)

Also not sure if we should use prometheus "protocol". 
My idea would be, a claymore api compatible.  there is already a monitor tool for that.

**Comment by chfast** [comment]
Yes, I'm working on bringing libmicrohttpd to Hunter, but it is not top priority. Do you need JSON RPC for this?

**Comment by smurfy** [comment]
Not necessarily, but libmicrohttpd would probably help in most cases. 
Will maybe do some more tests with the tcp socket server. i'm new to cmake and hunter.

**Comment by michael-pesce** [comment]
This would be HUGE! I'd love to be able to use Ethminer in AwesomeMiner with claymore-level monitoring!!!

**Comment by smurfy** [comment]
I will create a PR in the next 24h with basic stats support. (Works with ethmon from claymore)
The PR will be a Work in progress. More information in the PR description once created.

**Comment by liskl** [comment]
Still would love the Prometheus http /metrics endpoint. For me that would be way more scalable in large infrastructure and homelabs.

**Comment by smurfy** [comment]
@liskl I hear you, the plan is to extend the api to support different protocols in the future. (like prometheus). I think it depends a bit on ``libmicrohttpd`` or similiar library supported by hunter.

**Comment by jean-m-cyr** [comment]
> I think it depends a bit on libmicrohttpd or similiar library supported by hunter.

Why not just boost::network::http ?

**Comment by DeadManWalkingTO** [comment]
Now it's better.
Try latest version and feedback please.
Thank you!

**Comment by AndreaLanfranchi** [comment]
Easy to implement now with new API server.
Need specs about metrics

**Comment by 8-Lambda-8** [comment]
#2398
I Implemented it into the API Server

**Comment by 8-Lambda-8** [comment]
Grafana dashboard with data from Prometheus:
![image](https://user-images.githubusercontent.com/31072022/147891560-682a40fc-fa39-4fd9-8445-6fe621ce02ff.png)
